internal/app/server/server.go

```diff
@@ -78,6 +80,15 @@ func StartServer(opts *config.Options,
 			return fmt.Errorf("error load private key: %w", err)
 		}
 	}
+
+	// we wan't check error twice (already checked in main)
+	dsn, _ := storage.ParseDSN(opts.Config.DatabaseDSN)
+	// start flushing metrics on timer
+	if opts.Config.StoreInterval > 0 && dsn.Scheme == "memory" && opts.Config.StoreFile != "" {
+		go func() {
+			FlushingMetrics(ctx, opts.Storage, opts.Config.StoreFile, opts.Config.StoreInterval)
+		}()
+	}
```

Comment:
```suggestion
	// start flushing metrics on timer
	wg := sync.WaitGroup{}
	if opts.Config.StoreInterval > 0 && dsn.Scheme == "memory" && opts.Config.StoreFile != "" {
		wg.Add(1)
		go func() {
			defer wg.Done()
			FlushingMetrics(ctx, opts.Storage, opts.Config.StoreFile, opts.Config.StoreInterval)
		}()
	}
```
куда `wg.Wait()` воткнуть, думаю, сам знаешь.